### ISSUE TITLE
Expose create federation on bouncer client

### ISSUE THREAD

**Issue by carmenfan** [3drepo/3drepobouncer#132]

**Comment by carmenfan** [comment]
Federation can now be created through this command:

> genFed          Generate a federation. (args: file [owner])

it requires a single argument (with an optional second to specify the owner of the federation), that points to a JSON file that dictates the details of the federation.

The JSON file is expected to be of this format:

````javascript
{
			database: "databaseName",
			project: "projectName",
			subProjects: [
							{
								"database": "databaseName", //If this is empty, assume federation's database
								"project" : "project",
								"revId"   : "UUID in string", //Optional, defaults as head of master
								"isRevId": true/false, //Optional if revId is empty (true = revision id, false = branch id)
								"transformation": [...] //4by4 matrix, optional. defaults to identity
							},
							{...},
							{...}				
						 ]
		 }
````

So example of the file:
![image](https://cloud.githubusercontent.com/assets/11945337/17143258/b9d933ae-534a-11e6-9a98-f55ae835a004.png)

**Comment by carmenfan** [comment]
@thenrikie using this branch you should now be able to modify the backend to send a message to create a federation. It literally works the same as importFile, apart from the message needs to be like this:
````
genFed <json file> username
````
The json file needs to be formatted as described above. failure will return error code 9, or the usuals if it's auth error/invalid arguments


The only other thing is whether we keep to the same queue but set a priority for fed creation, or have 2 different queues all together, maybe even running a compute node instance on bouncer JS node just to process gen fed?. @timscully1983 discussion for tomorrow perhaps?

**Comment by carmenfan** [comment]
merged into v1.3
